### PR TITLE
Apply upstream changes to outlying files

### PR DESCRIPTION
**Pull request by ffoulkes**
- Replace CHECK_RETURN_IF_FALSE() with RET_CHECK().
- Address gcc compiler warnings in chassis_config_migrator.cc.

### CODE REVIEW

**Review by nupuruttarwar — APPROVED** [review]
LGTM

File: stratum/hal/config/chassis_config_migrator.cc [hunk 1]
```diff
@@ -31,6 +32,10 @@ ls -1 stratum/hal/config/*/chassis_config.pb.txt | \
     -chassis_config_file
 )USAGE";
 
+// Suppress gcc "no previous declaration" warnings.
```
Comment (5abeel) [inline stratum/hal/config/chassis_config_migrator.cc:35]: Not sure if we need this comment. Can get rid of it

**Review by 5abeel — APPROVED** [review]
LGTM. Just one minor comment about the comment

File: stratum/hal/config/chassis_config_migrator.cc (hunk 1 above)
Comment (ffoulkes) [reply stratum/hal/config/chassis_config_migrator.cc:35]: The comment is there to explain why there are prototypes for local functions.
It provides the non-obvious answer to the question, "why?"

**Pull request merged**